### ISSUE TITLE
`nix profile install foo` should override pre-installed `foo` by default

### ISSUE THREAD

**Issue by peterbecich** [NixOS/nix#11702]
**Is your feature request related to a problem? Please describe.**
This is about `nix profile`.

If `foo` is already installed via `nix profile`, attempting to re-install or update with `nix profile install foo` will result in error:
```
warning: 'foo' is already installed
```
The workaround is `nix profile install foo --priority -1` or lower than the current `foo`.

Example: updating `direnv` installed via `nix profile`:
```
$ nix profile install nixpkgs#direnv
warning: 'direnv' is already installed
```

**Describe the solution you'd like**
`nix profile install ...` should override the current installation when the current installation is a lower version.  Or maybe override by default regardless of version.

**Describe alternatives you've considered**

**Additional context**

**Priorities**

Add :+1: to [issues you find important](https://github.com/NixOS/nix/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

**Comment by bouk** [comment]
I agree, installing a package with the same name should 'upgrade' it by default
